pkg/service/backup/service_backup_restore_integration_test.go

```diff
@@ -362,40 +381,55 @@ func TestRestoreSchemaSmokeIntegration(t *testing.T) {
 		testParallel  = 1 // Restoring schema can't be done in parallel
 	)
 
-	target := RestoreTarget{
-		Location: []Location{
-			{
-				DC:       "dc1",
-				Provider: S3,
-				Path:     testBucket,
-			},
+	location := []Location{
+		{
+			DC:       "dc1",
+			Provider: S3,
+			Path:     testBucket,
 		},
-		Keyspace:      []string{"system_schema"},
-		BatchSize:     testBatchSize,
-		Parallel:      testParallel,
-		RestoreSchema: true,
 	}
+	locStr, err := json.Marshal(location)
+	if err != nil {
+		t.Fatal(err)
+	}
+	schemaTargetProps := fmt.Sprintf(`
+{
+  "location": %s,
+  "snapshot_tag": "sm_20060102150405UTC",
+  "batch_size": %d,
+  "parallel": %d,
+  "restore_schema": true
+}`, locStr, testBatchSize, testParallel)
 
-	smokeRestore(t, target, testKeyspace, testLoadCnt, testLoadSize)
+	smokeRestore(t, location[0], schemaTargetProps, testKeyspace, testLoadCnt, testLoadSize, true)
 }
 
-func smokeRestore(t *testing.T, target RestoreTarget, keyspace string, loadCnt, loadSize int) {
+func smokeRestore(t *testing.T, location Location, targetStr string, keyspace string, loadCnt, loadSize int, dropKeyspaces bool) {
 	var (
 		ctx          = context.Background()
 		cfg          = DefaultConfig()
 		srcClientCfg = scyllaclient.TestConfig(ManagedSecondClusterHosts(), AgentAuthToken())
 		mgrSession   = CreateScyllaManagerDBSession(t)
-		dstSession   = CreateSessionAndDropAllKeyspaces(t, ManagedClusterHosts())
-		srcSession   = CreateSessionAndDropAllKeyspaces(t, ManagedSecondClusterHosts())
-		dstH         = newRestoreTestHelper(t, mgrSession, cfg, target.Location[0], nil)
-		srcH         = newRestoreTestHelper(t, mgrSession, cfg, target.Location[0], &srcClientCfg)
+		dstH         = newRestoreTestHelper(t, mgrSession, cfg, location, nil)
+		srcH         = newRestoreTestHelper(t, mgrSession, cfg, location, &srcClientCfg)
+		dstSession   gocqlx.Session
+		srcSession   gocqlx.Session
```

Comment:
Maybe it's worth to leave the basic `restoreTablesSmokeIntegration` test and write a brand new one for this case? This would perhaps allow us to avoid the ifology in `smokeRestore` function in favor of creating another one.

Reply:
I don't think it make sense to leave previous restoreTablesSmokeIntegration.
It did almost the same what the current one does.
The difference is that the current one restores the schema the "universal restore" way + current one creates materialized view + secondary index in backup.